### PR TITLE
API V2 cart_controller#empty endpoint

### PR DESCRIPTION
**Pull request by WitekL**

### CODE REVIEW

File: api/app/controllers/spree/api/v2/storefront/cart_controller.rb [hunk 1]
```diff
@@ -24,6 +24,17 @@ def add_item
             end
           end
 
+          def empty
+            if spree_current_order.nil?
+              render json: { error: "Order doesn't exist" }, status: 404
```
Comment (damianlegawiec) [inline api/app/controllers/spree/api/v2/storefront/cart_controller.rb]: @WitekL let's just raise `ActiveRecord::RecordNotFound` here, ok?

File: api/spec/requests/spree/api/v2/storefront/cart_spec.rb [hunk 2]
```diff
@@ -100,4 +100,44 @@
...
+    end
+
+    context 'with existing guest order and line item' do
+      let(:guest_token) { 'guest_token' }
+      let!(:order) { create(:order, guest_token: guest_token) }
+      let!(:line_item) { create(:line_item, order: order) }
+
+      it 'empites the order' do
+        headers = { 'Authorization' => "Bearer #{token.token}" }
+        post '/api/v2/storefront/cart/empty', params: { order_token: guest_token }, headers: headers
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/cart_spec.rb]: @WitekL let's not use `params: { order_token: guest_token }` - it should be passed as `X-Spree-Order-Token` header, also for the guest checkout flow - we shouldn't pass the `Bearer` header at all

File: api/spec/requests/spree/api/v2/storefront/cart_spec.rb [hunk 3]
```diff
@@ -100,4 +100,44 @@
...
+      let!(:order) { create(:order, user: user) }
+      let!(:line_item) { create(:line_item, order: order) }
+
+      it 'empties the order' do
+        headers = { 'Authorization' => "Bearer #{token.token}" }
+        post '/api/v2/storefront/cart/empty', headers: headers
+
+        expect(response.status).to eq(200)
+
+        expect(order.line_items.count).to eq(0)
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/cart_spec.rb]: @WitekL please use shared examples here - those tests for signed in user and guest user are basically the same

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: api/app/controllers/spree/api/v2/storefront/cart_controller.rb (hunk 1 above)
Comment (kushniryb) [reply api/app/controllers/spree/api/v2/storefront/cart_controller.rb]: We should also start utilising `i18n` 🙄

File: api/config/routes.rb [hunk 4]
```diff
@@ -130,6 +130,7 @@
       namespace :storefront do
         resource :cart, controller: :cart, only: [:show, :create] do
           post 'add_item'
+          post 'empty'
```
Comment (kushniryb) [inline api/config/routes.rb]: Can we use symbols here?

File: api/config/routes.rb [hunk 5]
```diff
@@ -130,6 +130,7 @@
       namespace :storefront do
         resource :cart, controller: :cart, only: [:show, :create] do
```
Comment (kushniryb) [inline api/config/routes.rb]: `%i[]`

**Review by kushniryb — COMMENTED** [review]
LGTM overall 👍

**Comment by WitekL** [comment]
@damianlegawiec Can you take a look?

**Review by damianlegawiec — APPROVED** [review]
👍

**Pull request merged**